src/lib.rs

```diff
@@ -0,0 +1,55 @@
+#[macro_use]
+extern crate helix;
+extern crate rand;
+
+use rand::{ChaChaRng, Rng, SeedableRng};
+use std::collections::HashMap;
+
+#[derive(Debug, Clone)]
+struct DataSource {
+    random: ChaChaRng,
+}
+
+ruby! {
+  class HypothesisCoreEngine {
+    struct {
+      random: ChaChaRng,
+      next_id: u64,
+      children: HashMap<u64, DataSource>,
+    }
+
+    def initialize(helix, seed: u64){
+      let xs: [u32; 2] = [seed as u32, (seed >> 32) as u32];
+      HypothesisCoreEngine{
+        helix,
+        random: ChaChaRng::from_seed(&xs),
+        next_id: 0,
+        children: HashMap::new(),
+      }
+    }
+
+    def new_source(&mut self) -> u64 {
+      let result = self.next_id;
+      self.children.insert(result, DataSource{random: self.random.clone()});
+      self.random.next_u64();
+      self.next_id += 1;
+      return result;
```

Comment:
minor: I believe implicit return is more idiomatic in Rust.

Reply:
return keyword should be used for early returns in error cases, and no return keyword and missing semicolon at the end of the function is quasi idiomatic.

Reply:
I did actually know this one, I just don't agree. 😄 

I think I'm going to stick to explicit returns (in both the rust and ruby sections of the code). I've long ago resigned myself to being slightly unidiomatic in most languages, and I find explicit returns much clearer in imperative languages, especially when the final semicolon or lack thereof is significant otherwise.